src/commands/rename.rs

```diff
@@ -0,0 +1,109 @@
+use super::*;
+use crate::core::Target;
+use clap::Arg;
+use tracing_batteries::prelude::*;
+use crate::git;
+
+pub struct RenameCommand;
+crate::command!(RenameCommand);
+
+#[async_trait]
+impl CommandRunnable for RenameCommand {
+    fn name(&self) -> String {
+        String::from("rename")
+    }
+
+    fn app(&self) -> clap::Command {
+        clap::Command::new(self.name())
+            .about("renames a repository on your local machine")
+            .long_about("This command will rename the specified repository on your local machine. It requires that the repository name be provided in fully-qualified form.")
+            .arg(Arg::new("repo")
+                .help("The name of the repository to rename.")
+                .long_help("The repository to be renamed in fully-qualified form.")
+                .index(1)
+                .required(true))
+            .arg(Arg::new("new_name")
+                .help("The new name of the repository.")
+                .long_help("The new name of the repository must not be in fully-qualified form.")
+                .index(2)
+                .required(true))
+            .arg(Arg::new("update-git-remote")
```

Comment:
Is there any scenario where we wouldn't want this to be the default behaviour?

Reply:
yeah, likely will invert the logic here to `--no-update-remote` or something in cases where you want to explicitly disable it. 
Also subject to change once I integrate the GitHub API to rename the repo there too